core/src/game_state.rs

```diff
@@ -1209,6 +1209,18 @@ impl PlayPhase {
             msgs.push(MessageVariant::BonusLevelEarned);
         };
 
+        let current_landlord_idx = bail_unwrap!(self
+            .propagated
+            .players
+            .iter()
+            .position(|p| p.id == self.landlord));
+
+        if landlord_won && self.propagated.players[current_landlord_idx].rank() == Number::Ace {
```

Comment:
I think we already create a message at the end of each game? Can we add a field to that, or see if it is already included?

Reply:
GameFinished message is for every game's result. It is only used to generate game finished message. All other messages on player advances are separately generated. I think it is much cleaner to have a separate message for Gameset winner, it makes ChatMessage handling much simpler